services/src/main/java/org/broadleafcommerce/payment/service/gateway/DefaultPayPalCheckoutTransactionService.java

```diff
@@ -97,12 +103,15 @@
     @Getter(AccessLevel.PROTECTED)
     private final RetryTemplate retryTemplate;
 
+    @Setter(onMethod_ = {@Autowired})
+    @Getter(AccessLevel.PROTECTED)
+    private PaymentResponseUtil paymentResponseUtil;
+
     @Override
     public PaymentResponse authorize(PaymentRequest paymentRequest) {
-        PaymentResponse paymentResponse =
-                new PaymentResponse(DefaultPaymentTypes.THIRD_PARTY_ACCOUNT,
-                        PayPalCheckoutPaymentGatewayType.PAYPAL_CHECKOUT)
-                                .transactionType(DefaultTransactionTypes.AUTHORIZE);
+        PaymentResponse paymentResponse = paymentResponseUtil.buildPaymentResponse(paymentRequest,
+                PayPalCheckoutPaymentGatewayType.PAYPAL_CHECKOUT,
+                DefaultTransactionTypes.AUTHORIZE);
```

Comment:
We should move the `paymentResponseUtil.buildPaymentResponse(...)` call inside of the `try` block & we should catch `Exception` instead of `PaymentException`, to guarantee that this method will always return a PaymentResponse